engine/render/src/render/material.cpp

```diff
@@ -243,6 +227,52 @@ namespace dmRender
         m->m_MaterialAttributeValues.SetCapacity(num_attribute_byte_size);
         m->m_MaterialAttributeValues.SetSize(num_attribute_byte_size);
         memset(m->m_MaterialAttributeValues.Begin(), 0, num_attribute_byte_size);
+
+        // Assign default values based on semantic type:
+        //   * position and tangent gets 0,0,0,1 if a vec4 is used
+        //   * color gets 1,1,1,1
+        //   * world and normal matrix gets a 1 in all diagonals
+        for (int i = 0; i < num_program_attributes; ++i)
```

Comment:
TODO:
I think we can remove the data backing for vertex attributes that isn't customized by the editor, since we produce vertex data based on the ruleset.